be2-scala/src/main/scala/ch/epfl/pop/json/MessageDataProtocol.scala

```diff
@@ -153,20 +153,22 @@ object MessageDataProtocol extends DefaultJsonProtocol {
             lao.convertTo[Hash],
             frontend.convertTo[PublicKey],
             address,
-            peers.map(_.convertTo[String]).toList
+            peers.map(jsValue => jsValue.asJsObject.getFields(PARAM_ADDRESS).map(_.convertTo[String])).toList.flatten
```

Comment:
You could have a slightly more meaningful error if you explicitly checked that each jsValue is a string (JsString) as expected :
```scala
peers.map(jsValue => jsValue.asJsObject.getFields(PARAM_ADDRESS) match {
  case JsString(address) => address
  case _ => throw new IllegalArgumentException(s"Can't parse json value $jsValue to get an address")
}).toList       
```

Reply:
Good point ! Fixed.